src/libzstd.jl

```diff
@@ -44,11 +64,27 @@ mutable struct CStream
     obuffer::OutBuffer
 
     function CStream()
-        ptr = LibZstd.ZSTD_createCStream()
+        default_custom_mem = ZSTD_customMem(
+            @cfunction(zstd_custom_alloc, Ptr{Cvoid}, (Ptr{Cvoid}, Csize_t)),
+            @cfunction(zstd_custom_free, Cvoid, (Ptr{Cvoid}, Ptr{Cvoid})),
+            C_NULL,
+        )
+        ptr = ccall(
+            (:ZSTD_createCStream_advanced, libzstd),
```

Comment:
`ZSTD_createCStream_advanced` is part of the experimental API though according to https://github.com/facebook/zstd/issues/1734 these functions should not change much in the future.

I am using it because the regular `ZSTD_createCStream` leads to Julia not running the finalizer before being killed by OOM.